test/uk/gov/hmrc/gform/core/TemplateValidatorSpec.scala

```diff
@@ -69,31 +73,29 @@ class TemplateValidatorSpec extends Spec {
     }
   }
 
-  "validateEnrolmentIdentifier" should
-    "validates ${user.enrolledIdentifier} with HmrcSimpleModule and HmrcAgentModule but invalid with Anonymous" in {
-    import FormTemplateValidator._
-    import AuthConfigGen._
-    import FormComponentGen._
+  it should "validates ${user.enrolledIdentifier} with HmrcSimpleModule and HmrcAgentModule" in {
+    forAll(FormTemplateGen.formTemplateGen, Gen.oneOf(hmrcEnrolmentModuleGen, hmrcAgentWithEnrolmentModuleGen)) {
+      (template, authConfig) =>
+        val newTemplate = template.copy(authConfig = authConfig)
+        val isAUserCtx = userContextComponentType(formTemplate.expandFormTemplate.allFCs)
 
-    forAll(
-      FormTemplateGen.formTemplateGen,
-      Gen.oneOf(hmrcEnrolmentModuleGen, hmrcAgentWithEnrolmentModuleGen),
-      formComponentGen()) { (template, authConfig, fc) =>
-      val componentType = Text(EORI, UserCtx(EnrolledIdentifier))
-      val newFormComponents: List[FormComponent] = fc.copy(`type` = componentType) :: Nil
-      val newSections = template.sections.map(_.copy(fields = newFormComponents))
-      val newTemplate = template.copy(sections = newSections).copy(authConfig = authConfig)
-
-      val isAUserCtx = userContextComponentType(formTemplate.expandFormTemplate.allFCs)
-
-      whenever(
-        authConfig.isInstanceOf[HmrcEnrolmentModule] || authConfig
-          .isInstanceOf[HmrcAgentWithEnrolmentModule] && isAUserCtx.nonEmpty) {
-        validateEnrolmentIdentifier(newTemplate) should be(Valid)
-      }
+        whenever(authConfig.isInstanceOf[HmrcEnrolmentModule] || authConfig.isInstanceOf[HmrcAgentWithEnrolmentModule]) {
+          validateEnrolmentIdentifier(newTemplate, _ => true) should be(Valid)
+        }
 
-      whenever(isAUserCtx.isEmpty) {
-        validateEnrolmentIdentifier(newTemplate) should be(Valid)
+        whenever(!isAUserCtx) {
+          validateEnrolmentIdentifier(newTemplate, userContextComponentType) should be(Valid)
+        }
+    }
+  }
+
+  it should "invalid with Anonymous" in {
```

Comment:
If we want to test Anonymous (ie. `anonymousGen`), we should check also for other cases of `AuthConfig`.